.github/workflows/build-and-publish.yml

```diff
@@ -0,0 +1,66 @@
+name: Build and publish artifacts
+
+on: push
+
+jobs:
+  build-artifacts:
```

Comment:
Building and publishing in separate steps so that we can ensure that each commit builds, but we only publish tagged commits.